packages/mdc-button/_mixins.scss

```diff
@@ -17,25 +17,120 @@
 @import "@material/ripple/mixins";
 @import "@material/theme/mixins";
 @import "@material/theme/variables";
+@import "./variables";
 
-@mixin mdc-button-filled_() {
-  @include mdc-ripple-base;
-  @include mdc-ripple-bg((pseudo: "::before", base-color: white, opacity: .32));
-  @include mdc-ripple-fg((pseudo: "::after", base-color: white, opacity: .32));
-  @include mdc-theme-prop(color, text-primary-on-dark);
+@mixin mdc-button-base_() {
+  @include mdc-typography(button);
 
-  background-color: black;
+  display: inline-block;
+  position: relative;
+  min-width: 88px;
+  height: $mdc-button-height;
+  padding: 0 16px;
+  border: none;
+  box-sizing: border-box;
+  outline: none;
+  text-align: center;
+  overflow: hidden;
+  vertical-align: middle;
+  user-select: none;
+  -webkit-appearance: none;
 
-  @include mdc-theme-dark(".mdc-button") {
-    @include mdc-ripple-base;
-    @include mdc-ripple-bg((pseudo: "::before", base-color: black, opacity: .32));
-    @include mdc-ripple-fg((pseudo: "::after", base-color: black, opacity: .32));
-    @include mdc-theme-prop(color, text-primary-on-light);
+  // postcss-bem-linter: ignore
+  &:active {
+    outline: none;
+  }
+
+  &:hover {
+    cursor: pointer;
+  }
+
+  &::-moz-focus-inner {
+    padding: 0;
+    border: 0;
+  }
+
+  &:disabled {
+    cursor: default;
+    pointer-events: none;
+  }
+}
+
+@mixin mdc-button--raised_() {
+  @include mdc-elevation(2);
+  @include mdc-elevation-transition;
+
+  &:hover,
+  &:focus {
+    @include mdc-elevation(4);
+  }
+
+  &:active {
+    @include mdc-elevation(8);
+  }
+
+  &:disabled {
+    @include mdc-elevation(0);
+  }
+}
+
+@mixin mdc-button--compact_() {
+  padding: 0 8px;
+}
+
+@mixin mdc-button--dense_() {
+  height: $mdc-dense-button-height;
+  font-size: .8125rem; // 13sp
+  line-height: $mdc-dense-button-height;
+}
+
+@mixin mdc-button-container-fill-color($color) {
+  @at-root #{selector-append("a", &)}, &:enabled {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-button-stroke-color($color) {
+  @at-root #{selector-append("a", &)}, &:enabled {
+    @include mdc-theme-prop(border-color, $color);
+  }
+}
+
+@mixin mdc-button-ink-color($color) {
+  @at-root #{selector-append("a", &)}, &:enabled {
+    @include mdc-theme-prop(color, $color);
+  }
+}
 
-    background-color: white;
+@mixin mdc-button-ripple($ripple-config) {
+  @at-root #{selector-append("a", &)}, &:enabled {
+    @include mdc-ripple-base;
+    @include mdc-ripple-bg(map-merge((pseudo: "::before"), $ripple-config));
+    @include mdc-ripple-fg(map-merge((pseudo: "::after"), $ripple-config));
 
     &:not(.mdc-ripple-upgraded) {
-      -webkit-tap-highlight-color: rgba(black, .18);
+      $tap-highlight-color: rgba(map-get($ripple-config, base-color), map-get($ripple-config, opacity));
```

Comment:
Is it really safe to assume that both base-color and opacity are set? mdc-ripple/_mixins.scss has defaults for these but here we have nothing to fall back on.

Reply:
I think that is a totally valid concern. 
My suggestion is when we implement state, we move this css fall back as part of ripple. At the end of the day, this is css only fallback for ripple, which might not belong to this place in the first place? What do you think?